neqo-transport/src/connection/mod.rs

```diff
@@ -2252,6 +2274,10 @@ impl Connection {
             let payload_start = builder.len();
             let (mut tokens, mut ack_eliciting, mut padded) = (Vec::new(), false, false);
             if let Some(ref close) = closing_frame {
```

Comment:
I was suggesting that the entirety of this block (not the inner one; down to 2265(left)/2291(right)) be moved to a `write_closing_frames` function or similar.  Maybe there are too many arguments for that to look neat, I don't know.